db/config.cc

```diff
@@ -850,6 +850,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
         "Number of threads with which to deliver hints. In multiple data-center deployments, consider increasing this number because cross data-center handoff is generally slower.")
     , batchlog_replay_throttle_in_kb(this, "batchlog_replay_throttle_in_kb", value_status::Unused, 1024,
         "Total maximum throttle. Throttling is reduced proportionally to the number of nodes in the cluster.")
+    , batchlog_replay_cleanup_after_replays(this, "batchlog_replay_cleanup_after_replays", value_status::Unused, 100,
+        "Clean up batchlog memtable after every N replays. Replays are issued on a timer, every ring_delay_ms milliseconds. So if batchlog_replay_cleanup_after_replays is set to 100, the batchlog memtable is flushed every 100 * ring_delay_ms milliseconds")
```

Comment:
ring_delay_ms is very low in tests (0) and is mostly meaningless. I think we should redefine the timer, and set this to also be a duration.

Reply:
Make it LiveUpdate too.

Reply:
The replay is issued every 60 seconds except the first replay during boot. Only the first replay sleeps ring_delay_ms. @denesb Can you double check and confirm?

```
satic constexpr uint32_t replay_interval = 60 * 1000; // milliseconds

   auto delay = _delay;
    while (!_stop.abort_requested()) {
        try {
            co_await sleep_abortable(delay, _stop);
        } catch (sleep_aborted&) {
            co_return;
        }
        try {
            auto cleanup = post_replay_cleanup::no;
            if (++replay_counter >= _replay_cleanup_after_replays) {
                replay_counter = 0;
                cleanup = post_replay_cleanup::yes;
            }
            co_await do_batch_log_replay(cleanup);
        } catch (seastar::broken_semaphore&) {
            if (_stop.abort_requested()) {
                co_return;
            }
            on_internal_error_noexcept(blogger, fmt::format("Unexcepted exception in batchlog reply: {}", std::current_exception()));
        } catch (...) {
            blogger.error("Exception in batch replay: {}", std::current_exception());
        }
        delay = std::chrono::milliseconds(replay_interval);
    } 

```

So I think it makes sense to set  batchlog_replay_cleanup_after_replays to be a number of replay. We can make replay_interval configurable as well in a follow up if it is needed.